timefold-solver-python-core/src/main/python/score/_group_by.py

```diff
@@ -993,6 +1013,120 @@ def to_sorted_map(key_mapper, value_mapper, merge_function_or_set_creator=None):
+            'BiConstraintCollector[A, B, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B], Balanced_], load_function: Callable[[A, B], int],
+                     initial_load_function: Callable[[A, B], int]) -> \
+            'BiConstraintCollector[A, B, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C], Balanced_]) -> \
+            'TriConstraintCollector[A, B, C, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C], Balanced_],
+                     load_function: Callable[[A, B, C], int]) -> \
+            'TriConstraintCollector[A, B, C, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C], Balanced_], load_function: Callable[[A, B, C], int],
+                     initial_load_function: Callable[[A, B, C], int]) -> \
+            'TriConstraintCollector[A, B, C, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C, D], Balanced_]) -> \
+            'QuadConstraintCollector[A, B, C, D, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C, D], Balanced_],
+                     load_function: Callable[[A, B, C, D], int]) -> \
+            'QuadConstraintCollector[A, B, C, D, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @overload  # noqa
+    @staticmethod
+    def load_balance(balanced_item_function: Callable[[A, B, C, D], Balanced_],
+                     load_function: Callable[[A, B, C, D], int],
+                     initial_load_function: Callable[[A, B, C, D], int]) -> \
+            'QuadConstraintCollector[A, B, C, D, Any, LoadBalance[Balanced_]]':
+        ...
+
+    @staticmethod
+    def load_balance(balanced_item_function, load_function=None, initial_load_function=None):
+        """
+        Returns a collector that takes a stream of items and calculates the unfairness measure from them.
+        The load for every item is provided by the load_function,
+        with the starting load provided by the initial_load_function.
+
+        When this collector is used in a constraint stream,
+        it is recommended to use a score type which supports real numbers.
+        This is so that the unfairness measure keeps its precision
+        without forcing the other constraints to be multiplied by a large constant,
+        which would otherwise be required to implement fixed-point arithmetic.
+
+        Parameters
+        ----------
+        balanced_item_function:
```

Comment:
Should have Callable type on the right `Callable[[ParameterTypes, ...], ReturnType]`